zookeeper/Dockerfile

```diff
@@ -3,33 +3,55 @@
 # Ignoring DL4006 globally because we inherit the SHELL from our base image
 # hadolint global ignore=DL4006
 
+# ==============================================================================
+# START ZOOKEEPER BUILDER
+
+# FROM stackable/image/java-base AS builder
 FROM stackable/image/java-base
 
 ARG PRODUCT
 ARG JMX_EXPORTER
-ARG RELEASE
-
-LABEL name="Apache ZooKeeper" \
-      maintainer="info@stackable.tech" \
-      vendor="Stackable GmbH" \
-      version="${PRODUCT}" \
-      release="${RELEASE}" \
-      summary="The Stackable image for Apache ZooKeeper." \
-      description="This image is deployed by the Stackable Operator for Apache ZooKeeper."
 
 RUN microdnf update && \
-    microdnf install tar gzip zip && \
+    microdnf install \
+    # For the apply_patches.sh script
+    git \
+    gzip \
+    maven \
+    tar \
+    zip \
+    # hostname is needed by the maven ant run plugin for the "set-hostname-property" step
+    hostname && \
     microdnf clean all
 
+# Copy patches and JMX config into the builder
 COPY --chown=stackable:stackable zookeeper/stackable /stackable
-COPY zookeeper/licenses /licenses
 
 USER stackable
 WORKDIR /stackable
 
-RUN curl --fail -L "https://repo.stackable.tech/repository/packages/zookeeper/apache-zookeeper-${PRODUCT}-bin.tar.gz" | tar -xzC . && \
-    ln -s "/stackable/apache-zookeeper-${PRODUCT}-bin" /stackable/zookeeper
+# We use indivual RUN statements instead of chaining multiple commands together
+# to get better debug output when running the builder.
```

Comment:
Yes, I did this in the past as well but that would cause our Github Actions thing to fail because images can get too big with all the intermediate stuff.

I'm afraid in the final version we may want it all in one and cleaned up afterwards.

Reply:
https://github.com/stackabletech/docker-images/pull/638/commits/942a7f4e291bc5c98cf54822ea1e947b1a16c26f